lib/services/role.go

```diff
@@ -1853,13 +1852,21 @@ func (set RoleSet) CheckAccessToRemoteCluster(rc types.RemoteCluster) error {
 
 	rcLabels := rc.GetMetadata().Labels
 
-	// For backwards compatibility, if there is no role in the set with labels and the cluster
-	// has no labels, assume that the role set has access to the cluster.
+	// For backwards compatibility, if there is no role in the set with label
+	// matchers and the cluster has no labels, assume that the role set has
+	// access to the cluster.
 	usesLabels := false
+outer:
 	for _, role := range set {
-		if len(role.GetClusterLabels(types.Allow)) != 0 || len(role.GetClusterLabels(types.Deny)) != 0 {
-			usesLabels = true
-			break
+		for _, cond := range []types.RoleConditionType{types.Allow, types.Deny} {
```

Comment:
nit: this loop is same as `labelMatchersUnset`?

Reply:
right, good catch, thanks